test/instrumentation/async-await.js

```diff
@@ -0,0 +1,35 @@
+'use stirct'
+
+var agent = require('../..').start({
+  appName: 'test',
+  captureExceptions: false
+})
+var ins = agent._instrumentation
+
+var test = require('tape')
+
+test(function (t) {
+  t.plan(4)
+  var t1 = ins.startTransaction()
+  a1(100).then(function (result) {
+    t.equal(result, 'success')
+    t.equal(ins.currentTransaction && ins.currentTransaction.id, t1.id)
+  })
+  var t2 = ins.startTransaction()
+  a1(50).then(function (result) {
+    t.equal(result, 'success')
+    t.equal(ins.currentTransaction && ins.currentTransaction.id, t2.id)
+  })
+})
+
+async function a1 (delay) {
+  return a2(delay)
+}
```

Comment:
This method seems a little redundant. Why not call straight to `a2`?

We could put `a1` to a bit of use, by testing the `await` keyword:

```js
async function a1 (delay) {
    const res = await a2(delay);
    return res.toUpperCase();
}
```

That will at least make sure there are no gotchas with `async/await`. But might not be necessary.

Reply:
I wanted to make sure we used the await keyword somewhere. As far as I know you don't need to specify it if you just return right away (in fact my linter complained about it), so I removed it, but I was hoping it was implicit. But you're right it's probably better to have it there and then just do something with the result as you suggest

Reply:
Fixed 😃